app/components/search/Refinements/OpenNowFilter.tsx

```diff
@@ -39,17 +43,12 @@ const OpenNowFilter = ({ currentRefinement, refine }) => {
         name="openNow"
         id="openNow"
         className={styles.refinementInput}
-        value={isActive}
+        value="openNow"
```

Comment:
I think whoever wrote this component originally misunderstood what the `value` attribute means on checkbox `<input>` elements. It doesn't control the checked state; that's what the [`checked`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#additional_attributes) is for. In general, across all different subtypes of `<input>` elements, the `value` attribute is supposed to hold the value associated with the field, e.g. on text fields, it's the user-entered text.

Checkboxes are a bit weird because they can be used to represent a field that could have multiple items selected simultaneously, like in the phrase "check all that apply". The way this is supposed to be implemented in HTML is that you would have [all checkboxes that are part of the same field share the same `name` attribute, but each have distinct `value` attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#handling_multiple_checkboxes). When the form is turned into data, such as with an old-school synchronous form submission, or with the newer [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData) API, you would get an array of all the values corresponding to checked checkboxes.

TypeScript actually correctly caught the error, since `value` has to be a string, and we were passing in a boolean. The value doesn't really matter for a single-checkbox field, and especially because we don't ever turn this field into data (we instead attach an `onChange` handler), so I probably could have just deleted the `value` attribute altogether, but I guess it just felt like "habit" for me to set a checkbox's value to match its name and id.

Reply:
Oh I see! That is interesting. I didn't know. I appreciate the explanation!